### PR TITLE
Make ipMin optional (shortcircuits)

### PR DESCRIPTION
**Pull request by colinepiloquet**
Signed-off-by: Coline PILOQUET <coline.piloquet@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Update

### CODE REVIEW

**Review by annetill — APPROVED** [review]

File: iidm/iidm-impl/src/test/java/com/powsybl/iidm/network/impl/IdentifiableShortCircuitTest.java [hunk 1]
```diff
@@ -44,6 +44,20 @@ public void testWithoutIp() {
...
+    @Test
+    public void testWithoutIpMin() {
+        Network network = EurostagTutorialExample1Factory.create();
+        VoltageLevel voltageLevel = network.getVoltageLevel("VLLOAD");
+        assertNotNull(voltageLevel);
+        voltageLevel.newExtension(IdentifiableShortCircuitAdder.class)
+                .withIpMax(1000)
+                .add();
+        IdentifiableShortCircuit identifiableShortCircuit = voltageLevel.getExtension(IdentifiableShortCircuit.class);
+        assertEquals(Double.NaN, identifiableShortCircuit.getIpMin(), 0);
```
Comment (miovd) [inline iidm/iidm-impl/src/test/java/com/powsybl/iidm/network/impl/IdentifiableShortCircuitTest.java:59]: Checking if a value is  `NaN`, we do: `assertTrue(Double.isNaN(value))`

**Review by miovd — COMMENTED** [review]
One remark in the test, otherwise, it is okay

**Review by miovd — APPROVED** [review]

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/extensions/IdentifiableShortCircuitXmlSerializer.java [hunk 2]
```diff
@@ -34,13 +34,13 @@ public IdentifiableShortCircuitXmlSerializer() {
...
         XmlUtil.writeDouble("ipMax", identifiableShortCircuit.getIpMax(), context.getWriter());
-        XmlUtil.writeDouble("ipMin", identifiableShortCircuit.getIpMin(), context.getWriter());
+        XmlUtil.writeOptionalDouble("ipMin", identifiableShortCircuit.getIpMin(), Double.NaN, context.getWriter());
     }
 
     @Override
     public IdentifiableShortCircuit read(I identifiable, XmlReaderContext context) throws XMLStreamException {
         double ipMax = XmlUtil.readDoubleAttribute(context.getReader(), "ipMax");
-        double ipMin = XmlUtil.readDoubleAttribute(context.getReader(), "ipMin");
+        double ipMin = XmlUtil.readOptionalDoubleAttribute(context.getReader(), "ipMin", Double.NaN);
```
Comment (miovd) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/extensions/IdentifiableShortCircuitXmlSerializer.java:43]: You can use `readOptionalDoubleAttribute(context.getReader(), "ipMin");`

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/extensions/GeneratorShortCircuitXmlSerializer.java [hunk 3]
```diff
@@ -40,7 +40,7 @@ public void write(GeneratorShortCircuit generatorShortCircuit, XmlWriterContext
 
     @Override
     public GeneratorShortCircuit read(Generator generator, XmlReaderContext context) throws XMLStreamException {
-        double directSubtransX = XmlUtil.readDoubleAttribute(context.getReader(), "directSubtransX");
+        double directSubtransX = XmlUtil.readOptionalDoubleAttribute(context.getReader(), "directSubtransX");
```
Comment (colinepiloquet) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/extensions/GeneratorShortCircuitXmlSerializer.java:43]: I actually just realized that there was a mistake in the Generator serializer. I do the fix here, is that ok?

**Pull request merged**